cmd/jsontoml/main_test.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+	"os"
+	"io/ioutil"
+	"runtime"
+)
+
+func expectBufferEquality(t *testing.T, name string, buffer *bytes.Buffer, expected string) {
+	output := buffer.String()
+	if output != expected {
+		t.Errorf("incorrect %s: \n%sexpected %s: \n%s", name, output, name, expected)
+		t.Log([]rune(output))
+		t.Log([]rune(expected))
+	}
+}
+
+func expectProcessMainResults(t *testing.T, input string, args []string, exitCode int, expectedOutput string, expectedError string) {
+	inputReader := strings.NewReader(input)
+	outputBuffer := new(bytes.Buffer)
```

Comment:
You should just be able to write

```go
var outputBuffer bytes.Buffer
```

as a `bytes.Buffer` needs no initialization.

Reply:
This doesn't work. This is the error I get:
```
github.com/pelletier/go-toml/cmd/jsontoml/main_test.go:28:27: cannot use outputBuffer (type bytes.Buffer) as type io.Writer in argument to processMain:
	bytes.Buffer does not implement io.Writer (Write method has pointer receiver)
github.com/pelletier/go-toml/cmd/jsontoml/main_test.go:28:27: cannot use errorBuffer (type bytes.Buffer) as type io.Writer in argument to processMain:
	bytes.Buffer does not implement io.Writer (Write method has pointer receiver)
github.com/pelletier/go-toml/cmd/jsontoml/main_test.go:30:26: cannot use outputBuffer (type bytes.Buffer) as type *bytes.Buffer in argument to expectBufferEquality
github.com/pelletier/go-toml/cmd/jsontoml/main_test.go:31:26: cannot use errorBuffer (type bytes.Buffer) as type *bytes.Buffer in argument to expectBufferEquality
FAIL	github.com/pelletier/go-toml/cmd/jsontoml [build failed]
```

I got away with fixing `expectBufferEquality` by passing the addresses like &outputBuffer. But for some reason new(bytest.Buffer) works as a `io.Writer` type and var outputBuffer bytest.Buffer doesn't. Also this code was referred from the existing test for `tomljson`

Reply:
No worries. The reason why it fails is because the methods of the `io.Writer` interface are implemented on the pointer type (`*bytes.Buffer`), which `new` creates. You would indeed need to pass around the pointer (`&outputBuffer`).

Either way is fine with him. That wasn't a request for change, more like a FYI.

Reply:
Great. Acknowledged.